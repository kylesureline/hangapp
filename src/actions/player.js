import database from '../firebase/firebase';

export const initializeGame = (word, guessesRemaining) => ({
  type: 'INIT_GAME',
  answer: word,
  guessesRemaining
});

export const guessLetter = (letter) => ({
  type: 'GUESS_LETTER',
  letter
});

export const updateGuessedWord = (wordArray) => ({
  type: 'UPDATE_GUESSED_WORD',
  wordArray
});

export const wrongGuess = () => ({
  type: 'WRONG_GUESS'
});

export const setPlayer = (player) => ({
  type: 'SET_PLAYER',
  player
});

export const setPlayerStats = (stats) => ({
  type: 'SET_PLAYER_STATS',
  stats
});

export const setPastGames = (pastGames) => ({
  type: 'SET_PAST_GAMES',
  pastGames
});

export const startSetPlayer = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;

    return database.ref(`players/${uid}`)
      .once('value')
      .then((snapshot) => {
        const pastGames = [];

        for(let game in snapshot.val().pastGames) {
          pastGames.push(snapshot.val().pastGames[game]);
        }

        const stats = snapshot.val().stats;

        dispatch(setPlayerStats(stats));
        dispatch(setPastGames(pastGames));
      });

  };
};
