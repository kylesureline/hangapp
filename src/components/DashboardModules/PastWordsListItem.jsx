import React from 'react';
import moment from 'moment';

export const PastWordsListItem = (props) => (
  <div className="past-games__list-item">{props.word}, {props.type}, {props.def}, {props.won ? 'won': 'lost'} on {moment(props.playedAt).format('MMMM Do, YYYY')}<br /><br /></div>
);

export default PastWordsListItem;
