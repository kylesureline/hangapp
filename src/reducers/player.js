const defaultGameState = {
  answer: {
    word: '',
    type: '',
    def: ''
  },
  guessedLetters: [],
  guessedWord: [],
  guessesRemaining: 0,
  stats: {
    wins: 0,
    losses: 0
  }
};

export default (state = defaultGameState, action) => {
  switch(action.type) {
    case 'GUESS_LETTER':
      return {
        ...state,
        guessedLetters: [...state.guessedLetters, action.letter]
      };
    case 'UPDATE_GUESSED_WORD':
      return {
        ...state
      };
    case 'WRONG_GUESS':
      return {
        ...state,
        guessesRemaining: state.guessesRemaining -= 1
      };
    case 'SET_PLAYER_STATS':
      return {
        ...state,
        stats: action.stats
      };
    case 'SET_PAST_GAMES':
      return {
        ...state,
        pastGames: action.pastGames
      };
    case 'SET_GUESSES_REMAINING':
      return {
        ...state,
        guessesRemaining: action.guessesRemaining
      }
    case 'CHOOSE_DEFAULT_WORD':
      return {
        ...state,
        answer: {
          word: 'hangapp',
          type: 'noun',
          def: 'A hangman game built as a React PWA by Kyle Scheuerlein'
        }
      };
    case 'CHOOSE_RANDOM_WORD':
      return {
        ...state,
        answer: action.answer
      }
    case 'ADD_WORD':
      return {
        ...state,
        pastGames: [
          ...state.pastGames,
          action.answer
        ]
      }
    case 'ADD_WIN':
      return {
        ...state,
        stats: {
          ...state.stats,
          wins: state.stats.wins + 1
        }
      }
    case 'ADD_LOSS':
      return {
        ...state,
        stats: {
          ...state.stats,
          losses: state.stats.losses + 1
        }
      }
    default:
      return state;
  }
};
