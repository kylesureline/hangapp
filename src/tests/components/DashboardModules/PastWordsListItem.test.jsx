import React from 'react';
import { shallow } from 'enzyme';
import { PastWordsListItem } from '../../../components/DashboardModules/PastWordsListItem.jsx';
import { player } from '../../fixtures/player';

test('should render PastWordsListItem correctly', () => {
  const wrapper = shallow(<PastWordsListItem {...player.pastGames[2]}/>);
  expect(wrapper).toMatchSnapshot();
});
