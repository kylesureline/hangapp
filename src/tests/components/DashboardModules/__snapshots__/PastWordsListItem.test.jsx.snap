// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PastWordsListItem correctly 1`] = `
<div
  className="past-games__list-item"
>
  begonias
  , 
  noun
  , 
  any of a large genus (Begonia of the family Begoniaceae, the begonia family) of tropical or subtropical herbs and shrubs that have asymmetrical leaves and are widely cultivated as ornamentals
  , 
  lost
   on 
  August 13th, 2019
</div>
`;
